lib/configurators/databases/aws.go

```diff
@@ -0,0 +1,656 @@
+	// Re-use instructions as step descriptions.
+	descriptions := a.Instructions()
+	var results []ExecutionResult
+
+	// Create/Update policy document.
+	policyArn, err := a.upsertPolicyDocument(a.policyName, policyDescription, a.policyDocument)
+	results = append(results, ExecutionResult{Description: descriptions[0].Description, Err: err})
+	if err != nil {
+		return results
+	}
+
+	// Create/Update policy boundary document.
+	policyBoundaryArn, err := a.upsertPolicyDocument(a.policyBoundaryName, policyDescription, a.policyBoundaryDocument)
+	results = append(results, ExecutionResult{Description: descriptions[1].Description, Err: err})
+	if err != nil {
+		return results
+	}
+
+	// Attach policies to the target.
+	err = a.attachPolicies(policyArn, policyBoundaryArn)
+	results = append(results, ExecutionResult{Description: descriptions[2].Description, Err: err})
+	if err != nil {
+		return results
+	}
+
+	return results
+}
+
+// build generates the policy documents and store them into the
+// `awsConfigurator` struct.
+func (a *awsConfigurator) build() error {
+	var err error
+
+	// When running the command in manual mode, we want to have zero dependency
+	// with AWS configurations (like awscli or environment variables), so that
+	// the user can run this command and generate the instructions without any
+	// pre-requisite.
+	if !a.flags.Manual {
+		stsClient, err := a.getAWSSTSClient()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		a.awsIdentity, err = awslib.GetIdentityWithClient(context.Background(), stsClient)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	// Define the target and target type.
+	a.target, a.targetType, err = policiesTarget(a.flags, a.awsIdentity)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Generate policies.
+	a.policyDocument, a.policyBoundaryDocument, err = buildPolicyDocuments(
+		a.targetType,
+		a.flags,
+		a.fileConfig,
+		a.awsIdentity,
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Generate formatted policies.
+	a.formattedPolicyDocument, err = formatPolicyDocument(a.policyDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	a.formattedPolicyBoundaryDocument, err = formatPolicyDocument(a.policyBoundaryDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// retrieveCurrentPolicy returns information from AWS about a policy. If the
+// policy doesn't have `teleport` tag and error will be returned.
+func (a *awsConfigurator) retrieveCurrentPolicy(policyName string) (string, []*iam.PolicyVersion, error) {
+	if a.awsIdentity == nil {
+		return "", nil, trace.NotFound("AWS identity not provided, unable to determine policy account ID")
+	}
+
+	policyArn := fmt.Sprintf("arn:aws:iam::%s:policy/%s", a.awsIdentity.GetAccountID(), policyName)
+	client, err := a.getAWSIAMClient()
+	if err != nil {
+		return "", nil, trace.Wrap(err)
+	}
+
+	getPolicyResp, err := client.GetPolicy(&iam.GetPolicyInput{PolicyArn: aws.String(policyArn)})
```

Comment:
Let's use XXXWithContext SDK methods and pass context object everywhere from the caller.